# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/FetchEventOpChild.cpp
# Commit: 8fec60d0c499
# Full Hash: 8fec60d0c499fdf429f4d11da6bea0c65aecb1d4
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-27 09:40:54
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Depends on D138250
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r fb0ac558803d -r 8fec60d0c499 dom/serviceworkers/FetchEventOpChild.cpp
--- a/dom/serviceworkers/FetchEventOpChild.cpp	Sun Feb 27 00:26:21 2022 +0000
+++ b/dom/serviceworkers/FetchEventOpChild.cpp	Sun Feb 27 00:26:22 2022 +0000
@@ -230,9 +230,19 @@
     mPreloadResponseReadyPromise
         ->Then(
             GetCurrentSerialEventTarget(), __func__,
-            [this](SafeRefPtr<InternalResponse> aInternalResponse) {
-              auto response =
-                  aInternalResponse->ToParentToParentInternalResponse();
+            [this](FetchServiceResponse&& aResponse) {
+              SafeRefPtr<InternalResponse> preloadResponse;
+              IPCPerformanceTimingData timingData;
+              nsString initiatorType;
+              nsString entryName;
+              Tie(preloadResponse, timingData, initiatorType, entryName) =
+                  std::move(aResponse);
+              ParentToParentResponseWithTiming response;
+              response.response() =
+                  preloadResponse->ToParentToParentInternalResponse();
+              response.timingData() = timingData;
+              response.initiatorType() = initiatorType;
+              response.entryName() = entryName;
               if (!mWasSent) {
                 // The actor wasn't sent yet, we can still send the preload
                 // response with it.